### ISSUE TITLE
ReferenceError: origin is not defined

### ISSUE THREAD

**Issue by oles** [denoland/deno#24125]
It seems there's a miss match between the implementation of `origin` and/or `globalThis.origin` and `lib.dom.d.ts`

```
$ deno eval "console.log(origin)"
error: Uncaught (in promise) ReferenceError: origin is not defined
console.log(origin)
```

![Screenshot from 2024-06-06 23-16-09](https://github.com/denoland/deno/assets/7634050/1e5fb42a-3145-4567-a3c2-30f6d762e8f8)

![Screenshot from 2024-06-06 23-16-36](https://github.com/denoland/deno/assets/7634050/6faca767-8c12-4e20-9645-b43f35917bf2)

`location` works as expected when a `--location` is and isn't specified

```
$ deno eval --location=http://localhost:3000 "console.log(location)"
Location {
  hash: "",
  host: "localhost:3000",
  hostname: "localhost",
  href: "http://localhost:3000/",
  origin: "http://localhost:3000",
  pathname: "/",
  port: "3000",
  protocol: "http:",
  search: ""
}
```

```
$ deno eval "console.log(location)"
undefined
```

But it doesn't fix the issue with `origin` if you do

```
$ deno eval --location=http://localhost:3000 "console.log(origin)"
error: Uncaught (in promise) ReferenceError: origin is not defined
console.log(origin)
```

---

```
$ deno --version
deno 1.44.1 (release, x86_64-unknown-linux-gnu)
v8 12.6.228.3
typescript 5.4.5
```

**Comment by bartlomieju** [comment]
That's correct, `origin` is not an available global. Not everything that is defined in `lib.dom.d.ts` is available in Deno - eg. `document` is not available either. What use case do you have for `origin`?

**Comment by oles** [comment]
I see, thanks for the clarification!

I had an issue where, after a refactor, an `origin` variable got lost, and I didn't notice until I ran my test - I didn't know that `origin` was a global, and I didn't expect it to cause Deno to throw an error.

Having an easy way to get the `origin` is handy though, and using `globalThis.location.origin`/`location.origin` to get it is fine when `globalThis.origin`/`origin` isn't possible/supported yet.

I'd ideally like to configure the `--location` from a `.env` file though :thinking:

**Comment by oles** [comment]
For anyone else experiencing a similar issue - I found the [no-undef](https://lint.deno.land/rules/no-undef) deno_lint rule to be very useful.
